Sources/Shout/SFTP.swift

```diff
@@ -110,6 +110,32 @@ public class SFTP {
     ///   - permissions: the file permissions to create the new file with; defaults to FilePermissions.default
     /// - Throws: SSHError if local file can't be read or upload fails
     public func upload(localURL: URL, remotePath: String, permissions: FilePermissions = .default) throws {
+        let data = try Data(contentsOf: localURL, options: .alwaysMapped)
+        try upload(data: data, remotePath: remotePath, permissions: permissions)
+    }
+    
+    /// Upload a file from the local device to the remote server
+    ///
+    /// - Parameters:
+    ///   - string: String to be uploaded as a file
+    ///   - remotePath: the location on the remote server whether the file should be uploaded to
+    ///   - permissions: the file permissions to create the new file with; defaults to FilePermissions.default
+    /// - Throws: SSHError if string is not valid or upload fails
+    public func upload(string: String, remotePath: String, permissions: FilePermissions = .default) throws {
+        guard let data = string.data(using: .utf8) else {
+            throw SSHError.genericError("Unable to convert string to utf8 data")
+        }
+        try upload(data: data, remotePath: remotePath, permissions: permissions)
+    }
+    
+    /// Upload a file from the local device to the remote server
```

Comment:
I think this should read "Upload data to a file on the remote server" or something like that